Changelog.md

```diff
@@ -14,6 +14,7 @@ Compiler Features:
 Bugfixes:
  * Commandline Interface: Disallow the following options outside of the compiler mode: ``--via-ir``,``--metadata-literal``, ``--metadata-hash``, ``--model-checker-show-unproved``, ``--model-checker-div-mod-no-slacks``, ``--model-checker-engine``, ``--model-checker-invariants``, ``--model-checker-solvers``, ``--model-checker-timeout``, ``--model-checker-contracts``, ``--model-checker-targets``.
  * Type Checker: Fix null dereference in `abi.encodeCall` type checking of free function.
+ * Type Checker: Fix `bad_cast` exception when `abi.encodeCall` received a tuple returned from a function.
```

Comment:
Looks like this is more general than that. Happens even in cases without a function call.

Also, in cases where we have an uncaught exception, I'd call it "compiler crash" because from user's standpoint that's what it is. The fact that it's a C++ exception is just an unimportant implementation detail.
```suggestion
 * Type Checker: Fix compiler crash when `abi.encodeCall` received a tuple expression instead of an inline tuple.
```